script.js

```diff
@@ -1,12 +1,156 @@
+
+  episodeList.forEach((episode, index) => {
+    // Get the episode name, season number, episode number, and combine them
+    const episodeName = episode.name;
+    const paddedSeasonNum = ("0" + episode.season).slice(-2);
+    const paddedEpisodeNum = ("0" + episode.number).slice(-2);
+    const episodeNameNum = `${episodeName} - S${paddedSeasonNum}E${paddedEpisodeNum}`;
+
+    // Create the episode card elements
+    const episodeImageSrc = episode.image.medium;
+    const episodeDescription = episode.summary;
+
+    const episodeCardHtml = document.createElement("div");
+    episodeCardHtml.classList.add("episode-card");
+    episodeCardHtml.id = "episode-card" + index;
+
+    const episodeNameNumHolderHtml = document.createElement("div");
+    episodeNameNumHolderHtml.classList.add("episode-name-num-holder");
+
+    // Creating a span inside our span element
+    const episodeNameNumTextHtml = document.createElement("span");
+    episodeNameNumTextHtml.classList.add("episode-name-num-text");
+    episodeNameNumTextHtml.textContent = episodeNameNum;
+
+    const imageContainerHtml = document.createElement("div");
+    imageContainerHtml.classList.add("episode-image-container");
+
+    // Get the episode image and description
+    const episodeImageHtml = document.createElement("img");
+    episodeImageHtml.classList.add("medium-img");
+    episodeImageHtml.src = episodeImageSrc;
+
+    const episodeDescriptionContainerHtml = document.createElement("div");
+    episodeDescriptionContainerHtml.classList.add(
+      "episode-description-container"
+    );
+
+    const episodeDescriptionTextHtml = document.createElement("p");
+    episodeDescriptionTextHtml.classList.add("episode-description-text");
+    episodeDescriptionTextHtml.innerHTML = episodeDescription;
+
+    // Update info about displayed episodes
+    searchResultDisplayHtml.textContent = `Displaying ${cardsContainerHtml.length}/${episodeList.length} episodes`;
+
+    // Add the episode card elements to the container
+    episodeNameNumHolderHtml.appendChild(episodeNameNumTextHtml);
+    episodeCardHtml.appendChild(episodeNameNumHolderHtml);
+    episodeCardHtml.appendChild(imageContainerHtml);
+    imageContainerHtml.appendChild(episodeImageHtml);
+    episodeDescriptionContainerHtml.appendChild(episodeDescriptionTextHtml);
+    episodeCardHtml.appendChild(episodeDescriptionContainerHtml);
+    cardsContainerHtml.appendChild(episodeCardHtml);
+    rootHtml.appendChild(cardsContainerHtml);
+  });
 }
 
-window.onload = setup;
+// level 200
+
+// function to search
+
+function displaySearchedEpisodes(inputValue) {
+  // creating a fresh array which only holds the episodes that match our search criteria
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    return (
+      episode.name.toLowerCase().includes(inputValue) ||
+      episode.summary.toLowerCase().includes(inputValue)
+    );
+  });
+
+  // Display searchedEpisodes
+  if (filteredEpisodes.length !== 0) {
```

Comment:
Here is a small issue, do not remember whose Idea was to display filtered episodes only if we have them. I think better to display what we have, if we have none - display none. What u think?

Reply:
That makes sense. We don't display if we don't have any episode. I will need to correct it.